specsim/simulator.py

```diff
@@ -596,9 +596,12 @@ def generate_random_noise(self, random_state=None):
             mean_electrons = (
                 output['num_source_electrons'] +
                 output['num_sky_electrons'] + output['num_dark_electrons'])
-            output['random_noise_electrons'] = (
-                random_state.poisson(mean_electrons) - mean_electrons +
-                random_state.normal(scale=output['read_noise_electrons']))
+            if use_poisson :
+                output['random_noise_electrons'] = (
+                    random_state.poisson(mean_electrons) - mean_electrons +
+                    random_state.normal(scale=output['read_noise_electrons']))
+            else :
+                output['random_noise_electrons'] = random_state.normal(scale=np.sqrt( mean_electrons*(mean_electrons>0) + output['read_noise_electrons']**2))
```

Comment:
Why did you add the `(mean_electrons>0)` since this is not necessary for the Poisson case?  `mean_electrons >= 0` should always be true here (or else there is a bug).  Are you concerned about `mean_electrons == 0`?

Reply:
this was just to make sure we don't add a negative variance. but it is always positive or null I remove this test.